### ISSUE TITLE
use of `!` with `instanceof` should require parentheses

### ISSUE THREAD

**Issue by phpnode** [jshint/jshint#1825]
There is no legitimate reason to use `!foo instanceof Foo`, it requires parens and must be written as `!(foo instanceof Foo)`. Currently jshint does not warn on this syntax even though it is *guaranteed* to be a bug/typo. 

For instance, I could not get jshint to reject the following:

```js
function Test () {}

var y;

if (!y instanceof Test) {
   Test(); 
}
```

#437 mentions this but is closed, so maybe this is a regression?

**Comment by julienw** [comment]
Is it as simple as adding the case with "instanceof" in https://github.com/jshint/jshint/blob/6bf7653ec8eedcb35ea6825fd8b5a0abd2a4410e/src/jshint.js#L1150 ?

If yes I can give it a try.

**Comment by rwaldron** [comment]
@julienw I'm not sure and I would also have to try to answer. Give it a shot!

**Comment by julienw** [comment]
Crap, you didn't fall in my evil trap. :)
PR https://github.com/jshint/jshint/pull/2237

**Comment by phpnode** [comment]
The fix has been merged in #1825 so I'm closing this, thanks @julienw
